packages/ra-ui-materialui/src/list/filter/FilterButton.tsx

```diff
@@ -112,6 +118,7 @@ export const FilterButton = (props: FilterButtonProps): JSX.Element => {
     if (
         hiddenFilters.length === 0 &&
         validSavedQueries.length === 0 &&
+        !hasDisplayedFilters &&
```

Comment:
So if there are no displayed filters, you don't render the button? In that case, the user will never be able to add a filter... I fail to understand the logic here.

Reply:
Can't believe I did this.